### PR TITLE
minor updates for Firefox Puppeteer documentation

### PR DESCRIPTION
**Pull request by galgeek**
Hi, @chmanchester and @whimboo!

I looked through the Firefox Puppeteer documentation today, learned a bit about Sphinx, and made some minor, mostly nitpicky updates, mainly to the files in the ui directory. Note that I tried to regularize punctuation within files, but not much across files.

I don't know whether this is worth reviewing and merging before Tuesday.

### CODE REVIEW

File: firefox_puppeteer/ui/windows.py [hunk 1]
```diff
@@ -421,14 +426,14 @@ def navbar(self):
...
         the back, forward and home buttons. It also contains the location bar.
 
-        See the :class:`~ui.navbar.NavBar` reference.
+        See the :class:`~ui.toolbars.NavBar` reference.
         """
 
     @use_class_as_property('ui.tabbar.Tabs')
     def tabbar(self):
         """
         Provides access to the tab bar. This is the toolbar containing all the
```
Comment (chmanchester) [inline firefox_puppeteer/ui/windows.py]: I think this should get moved up a line, right after """

**Comment by chmanchester** [comment]
This looks like an improvement to me (and fixes a number of my typos and omissions). @galgeek, if you'd like to file a bug I will mark r+ there and get this merged. Thanks!

**Comment by galgeek** [comment]
Bug filed: https://bugzilla.mozilla.org/show_bug.cgi?id=1128654

I'm glad to know it's helpful. Thanks!

File: firefox_puppeteer/docs/conf.py [hunk 2]
```diff
@@ -13,7 +13,6 @@
 # serve to show the default.
 
 import os
-
```
Comment (whimboo) [inline firefox_puppeteer/docs/conf.py]: Please lets leave this in.

File: firefox_puppeteer/ui/tabbar.py [hunk 3]
```diff
@@ -20,7 +20,7 @@ class Tabs(BaseLib):
     @property
     def newtab_button(self):
         """
-        :returns: The new tab button element.
+        :returns: The new tab button element
```
Comment (whimboo) [inline firefox_puppeteer/ui/tabbar.py]: All those changes in this module will be invalid once I landed my big tabbar refactoring patch.

File: firefox_puppeteer/ui/toolbars.py [hunk 4]
```diff
@@ -41,10 +41,10 @@ def home_button(self):
 
     @use_class_as_property('ui.toolbars.LocationBar')
     def locationbar(self):
-        """Provides members for accessing and manipulationg the
+        """Provides members for accessing and manipulating the
         locationbar.
 
-        See the :class:`~ui.toolbars.LocationBar` reference.
+        See the :class:`LocationBar` reference.
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: Why this change? Does it still find the class?

Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: Oh, it should because its in the same module.

File: firefox_puppeteer/ui/toolbars.py [hunk 5]
```diff
@@ -114,7 +114,7 @@ def focus(self, evt='click'):
         """Focus the location bar according to the provided event.
 
         :param evt: The event to synthesize in order to focus the urlbar
-                    (one of 'click' or 'shortcut').
+                    (one of ``click`` or ``shortcut``).
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: A single tick please.

File: firefox_puppeteer/ui/toolbars.py [hunk 6]
```diff
@@ -188,7 +188,7 @@ def reload_url(self, trigger='button', force=False):
         """Reload the currently open page.
 
         :param trigger: The event type to use to cause the reload. (one of
-                        "shortcut", "shortcut2", or "button").
+                        ``shortcut``, ``shortcut2``, or ``button``).
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: Same here.

File: firefox_puppeteer/ui/toolbars.py [hunk 7]
```diff
@@ -267,11 +267,11 @@ def close(self, force=False):
             lambda _: not self.is_open)
 
     def get_matching_text(self, result, match_type):
-        """Retuns an array of strings of the matching text within a autocomplete
+        """Returns an array of strings of the matching text within an autocomplete
         result in the urlbar.
 
         :param result: The result to inspect for matches.
-        :param match_type: The type of match to search for (one of "title", "url").
+        :param match_type: The type of match to search for (one of ``title`` or ``url``).
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: And here.

File: firefox_puppeteer/ui/toolbars.py [hunk 8]
```diff
@@ -347,22 +347,42 @@ class IdentityPopup(BaseLib):
 
     @property
     def box(self):
+        """ Provides access to the identity-box element.
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: I would suggest we update it according to the doc comments I added for tabs.

File: firefox_puppeteer/ui/windows.py [hunk 9]
```diff
@@ -188,7 +188,7 @@ def __eq__(self, other):
 
     @property
     def closed(self):
-        """Returns closed state of the chrome window
+        """Returns closed state of the chrome window.
```
Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: For some you removed the dot, and here you add it. I would suggest we always keep it.

File: firefox_puppeteer/ui/windows.py [hunk 10]
```diff
@@ -324,20 +324,25 @@ def open_window(self, callback=None, expected_window_class=None):
...
+        """Sends a keyboard shortcut to the window.
 
-        :param command_key: The key (usually a letter) to be pressed
+        :param command_key: The key (usually a letter) to be pressed.
 
-        :param accel: Optional, If `True` the `Accel` modifier key is pressed. This key
-         differs between OS X (`Meta`) and Linux/Windows (`Ctrl`). Defaults to `False`
+        :param accel: Optional, If `True`, the `Accel` modifier key is pressed.
+                      This key differs between OS X (`Meta`) and
+                      Linux/Windows (`Ctrl`). Defaults to `False`.
```
Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: Why did you re-format all those param blocks? Is it just for in-code readability? Nothing will change for Sphinx here.

File: firefox_puppeteer/ui/tabbar.py (hunk 3 above)
Comment (galgeek) [inline firefox_puppeteer/ui/tabbar.py]: I'll checkout the master for tabbar.py and resubmit.

File: firefox_puppeteer/ui/windows.py (hunk 9 above)
Comment (galgeek) [inline firefox_puppeteer/ui/windows.py]: I tried to add periods to sentences and verb phrases consistently throughout and otherwise tried to use periods consistently within each file, aiming for a minimal review.

File: firefox_puppeteer/ui/toolbars.py (hunk 5 above)
Comment (galgeek) [inline firefox_puppeteer/ui/toolbars.py]: Restructured Text simply italicizes items enclosed with a single tick, while Markdown treats them as inline code. But I'm not entirely sure what's right here, so I'll defer to your suggestion.

File: firefox_puppeteer/ui/toolbars.py (hunk 8 above)
Comment (galgeek) [inline firefox_puppeteer/ui/toolbars.py]: My understanding is that you'd like to remove the text outside of the returns statement, since it is redundant here. That's what I've done.

File: firefox_puppeteer/ui/windows.py (hunk 10 above)
Comment (galgeek) [inline firefox_puppeteer/ui/windows.py]: The additional punctuation in the first line made it longer than Travis will accept.

File: firefox_puppeteer/docs/conf.py (hunk 2 above)
Comment (galgeek) [inline firefox_puppeteer/docs/conf.py]: Oops!

Deleted accidentally while I was attempting to persuade Sphinx to generate a module index.

File: firefox_puppeteer/ui/tabbar.py (hunk 3 above)
Comment (whimboo) [inline firefox_puppeteer/ui/tabbar.py]: Barbara Miller wrote on 02/03/2015 12:36 AM:
>>      def newtab_button(self):
>>          """
>> -        :returns: The new tab button element.
>> +        :returns: The new tab button element
> 
> I'll checkout the master for tabbar.py and resubmit.

The patch has been landed. So please feel free to check and update. Thanks

File: firefox_puppeteer/ui/windows.py (hunk 9 above)
Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: Please see for example https://github.com/mozilla/firefox-ui-tests/pull/58/files#diff-296ca2194a2c0e54996caffeb9bbd8baL14 where you have removed the dots.

File: firefox_puppeteer/ui/toolbars.py (hunk 5 above)
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: Barbara Miller wrote on 02/03/2015 01:04 AM:

>> -                    (one of 'click' or 'shortcut').
>> +                    (one of ``click`` or ``shortcut``).
> 
> Restructured Text simply italicizes items enclosed with a single tick,
> while Markdown treats them as inline code. But I'm not entirely sure
> what's right here, so I'll defer to your suggestion.


This is not markdown but sphinx. So when you do the changes use single
ticks. As an example have a look at
http://firefox-puppeteer.readthedocs.org/en/latest/ui/tabbar.html.

Also make sure that the `make html` command runs successfully under
firefox_puppeteer/docs.

File: firefox_puppeteer/ui/toolbars.py (hunk 8 above)
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: It's not redundant but it will cause broken documentation. The function will be marked to not return any value if you remove the `:returns:` line.

File: firefox_puppeteer/ui/windows.py (hunk 10 above)
Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: Then just make the line shorter, but keep the indentation. When you run the 'make html' command you will most likely see failures with this formatting.

File: firefox_puppeteer/ui/windows.py (hunk 9 above)
Comment (galgeek) [inline firefox_puppeteer/ui/windows.py]: I don't see any sentences or verb phrases there lacking periods, and in other places in the file, noun phrases don't use periods, so I tried to make those consistent.

Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: Please compare closely! Here you add the final period, while at the reference you remove it. I would suggest to always have a final period.

**Comment by galgeek** [comment]
It would be good to get these minor updates merged: I've verified using `make html` that Sphinx can build this set, I've added periods everywhere a person might want a period in firefox_puppeteer/ui/windows.py, and Travis like them, too.

File: firefox_puppeteer/ui/toolbars.py [hunk 11]
```diff
@@ -324,16 +324,15 @@ def is_open(self):
...
         return self.marionette.find_element(By.ID,
                                             'PopupAutoCompleteRichResult')
 
     @property
     def results(self):
-        """ Povides access to the container node for autocomplete results.
-
-        :returns: The result container node.
+        """
+        :returns: The autocomplete result container node
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: Have you looked at the properties I was referring to in the new tabbar and tab class? It's not mirroring it. I feel we should have a description and return line, and that the description tells about the DOM element.

File: firefox_puppeteer/ui/windows.py [hunk 12]
```diff
@@ -21,8 +21,8 @@ class Windows(BaseLib):
     def all(self):
         """Retrieves a list of all open chrome windows.
 
-        :returns: List of :class:`BaseWindow`'s corresponding to the
-                  windows in `marionette.chrome_window_handles`.
+        :returns: List of :class:`BaseWindow` instances corresponding to the
+                  windows in ``marionette.chrome_window_handles``.
```
Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: There are still two back ticks in use.

File: firefox_puppeteer/ui/windows.py [hunk 13]
```diff
@@ -129,11 +129,11 @@ def focus(self, handle):
     def switch_to(self, target):
         """Switches context to the specified chrome window.
 
-        :param target: The window to switch to. `target` can be a `handle` or a
+        :param target: The window to switch to. ``target`` can be a `handle` or a
```
Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: Same here.

File: firefox_puppeteer/ui/windows.py [hunk 14]
```diff
@@ -369,12 +373,12 @@ def send_shortcut(self, command_key, **kwargs):
     def switch_to(self, focus=False):
         """Switches the context to this chrome window.
 
-        By default it will not focus the window. If that behavior is wanted the
-        `focus` parameter can be used.
+        By default it will not focus the window. If that behavior is wanted, the
+        ``focus`` parameter can be used.
```
Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: Same here for back ticks.

File: firefox_puppeteer/ui/windows.py [hunk 15]
```diff
@@ -422,18 +426,16 @@ def is_private(self):
...
+        See the :class:`~ui.toolbars.NavBar` reference.
         """
 
     @property
     def tabbar(self):
-        """
-        Provides access to the tab bar. This is the toolbar containing all the
-        tabs, the new tab button, and the tab menu
+        """Provides access to the tab bar. This is the toolbar containing all the
+        tabs, the new tab button, and the tab menu.
```
Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: The description should fit into the first line. I don't think that we need the second sentence, so lets remove it.

File: firefox_puppeteer/ui/windows.py [hunk 16]
```diff
@@ -448,9 +450,9 @@ def tabbar(self):
     def close(self, trigger='menu', force=False):
         """Closes the current browser window by using the specified trigger.
 
-        :param trigger: Optional, method in how to close the current browser window. This can
-         be a string with one of `menu` or `shortcut`, or a callback which gets triggered
-         with the current :class:`BrowserWindow` as parameter. Defaults to `menu`.
+        :param trigger: Optional, method to close the current browser window. This can
+         be a string with one of ``menu`` or ``shortcut``, or a callback which gets triggered
+         with the current :class:`BrowserWindow` as parameter. Defaults to ``menu``.
```
Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: still double back ticks here.

**Comment by galgeek** [comment]
Hi @whimboo and @chmanchester!

I’ve updated per Henrik’s most recent feedback as best I can.

File: firefox_puppeteer/ui/toolbars.py [hunk 17]
```diff
@@ -11,55 +11,53 @@
...
+    """Provides access to the DOM elements contained in the
+    navigation bar as well as the location bar."""
 
     @property
     def back_button(self):
-        """ Provides access to the back button from the navbar ui.
+        """Provides access to the DOM element back button in the navbar.
 
-        :returns: The back button element.
+        :returns: Reference to the back button.
```
Comment (whimboo) [inline firefox_puppeteer/ui/toolbars.py]: Both changes are not what we have in the tabs module and for the identity class, but those things we could fix later if really necessary.

File: firefox_puppeteer/ui/windows.py (hunk 15 above)
Comment (whimboo) [inline firefox_puppeteer/ui/windows.py]: This is not consistent yet, but nothing which blocks us now. It can still be done later if necessary.

**Comment by whimboo** [comment]
This PR has been merged as https://github.com/mozilla/firefox-ui-tests/commit/48827d8331bab07ce0eafe819602c928af823486
